scripts/schema/loader.py

```diff
@@ -171,6 +171,26 @@ def nest_fields(field_array):
     return schema_root
 
 
+def array_of_dicts_to_set(array_vals):
+    ret_set = set()
+    for dict_val in array_vals:
+        ret_set.add(frozenset(dict_val.items()))
+    return ret_set
+
+
+def set_of_sets_to_array(set_vals):
+    ret_list = []
+    for set_info in set_vals:
+        ret_list.append(dict(set_info))
+    return sorted(ret_list, key=lambda k: k['name'])
+
+
+def dedup_and_merge_lists(list_a, list_b):
+    list_a_set = array_of_dicts_to_set(list_a)
+    list_b_set = array_of_dicts_to_set(list_b)
+    return set_of_sets_to_array(list_a_set | list_b_set)
```

Comment:
Minor issue I stumbled across while testing this out. Not sure it would be a blocker to merging, but worth noting the behavior.

The union will remove exact duplicate items:

```
> list_a_set
{frozenset({('name', 'text'), ('type', 'text')})}

> list_b_set
{frozenset({('name', 'text'), ('type', 'text')}), frozenset({('type', 'keyword'), ('normalizer', 'lowercase'), ('name', 'caseless')})}

> list_a_set | list_b_set
{frozenset({('name', 'text'), ('type', 'text')}), frozenset({('type', 'keyword'), ('normalizer', 'lowercase'), ('name', 'caseless')})}
```

But if the sets are not exact duplicates, it could lead to duplicate field names:

```
> list_a_set
{frozenset({('type', 'text'), ('name', 'text')})}

> list_b_set
{frozenset({('normalizer', 'lowercase'), ('type', 'keyword'), ('name', 'caseless')}), frozenset({('type', 'keyword'), ('name', 'text')})}

> list_a_set | list_b_set
{frozenset({('normalizer', 'lowercase'), ('type', 'keyword'), ('name', 'caseless')}), frozenset({('type', 'text'), ('name', 'text')}), frozenset({('type', 'keyword'), ('name', 'text')})}
```

schema include file:

```yaml
---
  - name: file
    title: File
    group: 2
    short: Fields describing files.
    description: >
      Custom file
    fields:
      - name: path
        multi_fields:
          - name: caseless
            type: keyword
            normalizer: lowercase
          - name: text  
            type: keyword <= I imagine this would only happen by accident 😃
```

Resulting intermediate state:

```yaml
        multi_fields:
        - flat_name: file.path.caseless
          ignore_above: 1024
          name: caseless
          normalizer: lowercase
          type: keyword
        - flat_name: file.path.text
          ignore_above: 1024
          name: text
          type: keyword
        - flat_name: file.path.text
          name: text
          norms: false
          type: text
```

Reply:
Oh good catch, what do we think the expected behavior should be in this scenario? I could put in a check to ensure that two of the same `name` fields don't exist in the resulting set and throw an error if they do? Or maybe just have core override?

Reply:
IMO we should dedupe on `name` and take the most recent definition in the case of dupes (this would allow for overrides).

Reply:
@webmat do you have any thoughts? I recall back in #864, logic was removed from the tooling to allow `--include` supplied custom fields to be more permissive:

> This means the tooling must now accept included files as they are, with all of the power this entails. 

Perhaps we simply make sure to note that users need to be aware of introducing such duplicates fields?

Reply:
I agree with @madirey. We should keep it simple and only ensure we have unique multi-field names.

The `--include` option is meant to override, so the ideal behaviour is for a custom multi-field definition to replace or be merged with an entry of the same name. I'm on the fence on whether to merge/replace an entry of the same name, though. Happy to be convinced either way.

But to take a concrete example, let's say someone has tuned an analyzer that works well for user agent strings, I want them to be able to replace the default `user_agent.original.text` multi-field with such a custom definition:

```YAML
        multi_fields:
        - name: text
          norms: false
          type: text
          normalizer: ua_analyzer 
```

I think I have a preference with merging the pre-existing multi-field definitions of the same name, as this is more in line with how everything else is handled with custom fields. And it has the bonus of allowing a more terse custom definition:

```YAML
        - name: text
          normalizer: ua_analyzer 
```